tests/hooks/useFieldArray.test.js

```diff
@@ -0,0 +1,429 @@
+import { createRef } from 'react';
+import { mount } from 'enzyme';
+import { act, render } from '@testing-library/react';
+import FieldArrayTestForm from '../testUtils/FieldArrayTestForm';
+import testHook from '../testUtils/hookTestUtils';
+import logger from '../../src/utils/logger';
+import { useForm, useFieldArray } from '../../src';
+import * as inputTypeUtils from '../../src/utils/inputTypeUtils';
+
+describe('useFieldArray hook', () => {
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('hook call result', () => {
+		it('should return multiple methods and attributes when called', () => {
+			let useFormResults;
+			let sut;
+
+			testHook(() => {
+				useFormResults = useForm();
+				sut = useFieldArray({ name: 'fieldArray', rules: {} }, useFormResults.formContext);
+			});
+
+			const expectedValues = {
+				fields: expect.any(Array),
+				register: expect.any(Function),
+				remove: expect.any(Function),
+				errors: {},
+			};
+
+			expect(sut).toMatchObject(expectedValues);
+		});
+	});
+
+	describe('form context provision check', () => {
+		jest.mock('react', () => {
+			const ActualReact = require.requireActual('react');
+
+			return { ...ActualReact, useContext: () => undefined };
+		});
+
+		it('should call logger.fatal if no context is provided to the hook', () => {
+			const initialNodeEnv = process.env.NODE_ENV;
+			process.env.NODE_ENV = 'test';
+
+			const loggerFatalSpy = jest.spyOn(logger, 'fatal');
+
+			testHook(() => {
+				expect(() => useFieldArray({ name: 'test-field-array' }, undefined)).toThrowError();
+			});
+
+			expect(loggerFatalSpy).toHaveBeenCalledTimes(1);
+
+			process.env.NODE_ENV = initialNodeEnv;
+		});
+	});
+
+	describe('registerField', () => {
```

Comment:
Isn't it `registerArrayField` being tested here ?

Reply:
The method is named `registerFIeld` in the useFieldArray hook. We use the `registerArrayField` denomination in useForm to mark the separation.